code/game/objects/items/melee/energy.dm

```diff
@@ -92,6 +92,9 @@
 	playsound(loc, hitsound, get_clamped_volume(), TRUE, -1)
 	add_fingerprint(user)
 
+/obj/item/melee/energy/get_demolition_modifier(obj/target)
+	return HAS_TRAIT(src, TRAIT_TRANSFORM_ACTIVE) ? demolition_mod : 1
```

Comment:
Chainsaws and fireaxes are also affected, either copy this to them (active/wielded respectively) or maybe make this a trait? Or make sharp objects not be affected if their default demo mod is above 1?